src/libasr/pass/print_list_tuple.cpp

```diff
@@ -0,0 +1,277 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/pass_utils.h>
+#include <libasr/pass/print_list_tuple.h>
+
+namespace LFortran {
+
+/*
+This ASR pass replaces print list with print every value,
+comma_space, brackets and newline. The function
+`pass_replace_print_list_tuple` transforms the ASR tree in-place.
+
+Converts:
+
+    print(l, sep="pqr", end="xyz") # l is a list
```

Comment:
The pass could be transformed to take `StringFormatPython1("%s", l)` and transform it into manual string creation.